sqlglot/dialects/tsql.py

```diff
@@ -132,6 +133,28 @@ def generate_date_delta_with_unit_sql(
     return self.func(func, expression.text("unit"), expression.expression, expression.this)
 
 
+def _parse_date_delta(
+    exp_class: t.Type[E], unit_mapping: t.Optional[t.Dict[str, str]] = None
+) -> t.Callable[[t.List], E]:
+    def inner_func(args: t.List) -> E:
+        unit_based = len(args) == 3
+        this = args[2] if unit_based else seq_get(args, 0)
+        unit = args[0] if unit_based else exp.Literal.string("DAY")
+        unit = exp.var(unit_mapping.get(unit.name.lower(), unit.name)) if unit_mapping else unit
+
+        start_date = str(seq_get(args, 1))
+        if start_date.isnumeric():
+            # perform date addition
+            days = int(start_date)
+            epoc = datetime.datetime.strptime("1900-01-01", "%Y-%M-%d")
```

Comment:
[Nit] This is a bit misleading, since "epoch" generally refers to 1970-01-01. I'm not sure what terminology T-SQL uses, but should we call it "start" or something else?

Reply:
change it to datetime.date(1900, 1, 1)